.github/workflows/release-nightly.yaml

```diff
@@ -1,227 +1,241 @@
+          - build: macos
+            os: macos-latest
+            target: x86_64-apple-darwin
+          - build: macos-arm64
+            os: macos-11
+            target: aarch64-apple-darwin
+          - build: windows-64bit
+            os: windows-latest
+            target: x86_64-pc-windows-msvc
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Update versions
+        shell: bash
+        run: |
+          version=0.0.0-nightly-$(date +"%y%m%d")
+          
+          sed "s/^version = \".*\"\$/version = \"$version\"/" compiler-core/Cargo.toml > /tmp/gleam-cargo.toml
+          mv /tmp/gleam-cargo.toml compiler-core/Cargo.toml
+          
+          sed "s/^version = \".*\"\$/version = \"$version\"/" compiler-wasm/Cargo.toml > /tmp/gleam-cargo.toml
+          mv /tmp/gleam-cargo.toml compiler-wasm/Cargo.toml
+          
+          sed "s/^version = \".*\"\$/version = \"$version\"/" compiler-cli/Cargo.toml > /tmp/gleam-cargo.toml
+          mv /tmp/gleam-cargo.toml compiler-cli/Cargo.toml
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          target: ${{ matrix.target }}
+          override: true
+          default: true
+          profile: minimal
+
+      - name: Handle Rust dependencies caching
+        uses: Swatinem/rust-cache@v1
+        with:
+          key: v1-${{ matrix.target }}
+
+      - name: Build release binary
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --target ${{ matrix.target }}
+
+      - name: Build archive
+        shell: bash
+        run: |
+          VERSION=nightly
+          DAY=$(date +"%y%m%d")
+
+          if [ "${{ matrix.os }}" = "windows-latest" ]; then
+            ARCHIVE="gleam-$VERSION-${{ matrix.build }}.zip"
+            cp "target/${{ matrix.target }}/release/gleam.exe" "gleam.exe"
+            7z a "$ARCHIVE" "gleam.exe"
+            rm gleam.exe
+          else
+            ARCHIVE="gleam-$VERSION-${{ matrix.build }}.tar.gz"
+            cp "target/${{ matrix.target }}/release/gleam" "gleam"
+            tar -czvf "$ARCHIVE" "gleam"
+            rm gleam
+          fi
+
+          openssl dgst -r -sha256 -out "$ARCHIVE".sha256 "$ARCHIVE"
+          openssl dgst -r -sha512 -out "$ARCHIVE".sha512 "$ARCHIVE"
+          echo "ASSET=$ARCHIVE" >> $GITHUB_ENV
+          echo "DAY=$DAY" >> $GITHUB_ENV
+
+      - name: Ensure binary successfully boots
+        shell: bash
+        run: |
+          if [ "${{ matrix.os }}" = "windows-latest" ]; then
+            7z x "$ASSET"
+            ./gleam.exe --version
+          elif [ "${{ matrix.build }}" = "macos-arm64" ]; then
+            echo We cannot test an ARM binary on a AMD64 runner
+          else
+            tar -xvzf "$ASSET"
+            ./gleam --version
+          fi
+
+      - name: Upload release archive
+        shell: bash
```

Comment:
How does this work if hub hasn't been installed yet?

Reply:
Hub is included with the standard github images.